lib/redis/event_handler.js

```diff
@@ -39,6 +39,25 @@ exports.connectHandler = function (self) {
 
     self.initParser();
 
+    self.lastWriteTime = null;
+    if (typeof self.options.timeoutPerCommand === 'number') {
+      var timeoutPerCommand = self.options.timeoutPerCommand;
+      debug('Per-command timeout set to %s', self.options.timeoutPerCommand);
+      self.timeoutPerCommand = setInterval(() => {
+        var commandQueueLength = self.commandQueue.length;
+        if (
+          commandQueueLength > 0 &&
+          Date.now() - self.lastWriteTime > self.options.timeoutPerCommand
+        ) {
+          debug('Command timed out. Pending commands: %s', commandQueueLength);
+          var err = new Error('Command timed out');
+          self.flushQueue(err, { offlineQueue: false });
+          self.silentEmit('error', err);
+          self.disconnect(true);
+        }
+      }, 500);
```

Comment:
It appears that the minimum practical value for `timeoutPerCommand` is 500ms.  Even if I specify `timeoutPerCommand = 100`, the check for a timed out command occurs every 500ms and so the command time out error won't trigger as quickly as I would expect.